crates/re_types/src/loggable.rs

```diff
@@ -0,0 +1,270 @@
+            None,
+        )
+    }
+
+    /// The underlying [`arrow2::datatypes::Field`], including datatype extensions.
+    ///
+    /// The default implementation will simply wrap [`Self::extended_arrow_datatype`] in a
+    /// [`arrow2::datatypes::Field`], which is what you want in most cases.
+    #[inline]
+    fn arrow_field() -> arrow2::datatypes::Field {
+        arrow2::datatypes::Field::new(
+            Self::name().to_string(),
+            Self::extended_arrow_datatype(),
+            false,
+        )
+    }
+
+    // --- Optional serialization methods ---
+
+    /// Given an iterator of owned or reference values to the current [`Loggable`], serializes
+    /// them into an Arrow array.
+    /// The Arrow array's datatype will match [`Loggable::arrow_field`].
+    ///
+    /// Panics on failure.
+    /// This will _never_ fail for Rerun's built-in [`Loggable`]s.
+    ///
+    /// For the fallible version, see [`Loggable::try_to_arrow`].
+    #[inline]
+    fn to_arrow<'a>(
+        data: impl IntoIterator<Item = impl Into<::std::borrow::Cow<'a, Self>>>,
+    ) -> Box<dyn ::arrow2::array::Array>
+    where
+        Self: 'a,
+    {
+        Self::try_to_arrow_opt(data.into_iter().map(Some)).detailed_unwrap()
+    }
+
+    /// Given an iterator of owned or reference values to the current [`Loggable`], serializes
+    /// them into an Arrow array.
+    /// The Arrow array's datatype will match [`Loggable::arrow_field`].
+    ///
+    /// This will _never_ fail for Rerun's built-in [`Loggable`].
+    /// For the non-fallible version, see [`Loggable::to_arrow`].
+    #[inline]
+    fn try_to_arrow<'a>(
+        data: impl IntoIterator<Item = impl Into<::std::borrow::Cow<'a, Self>>>,
+    ) -> SerializationResult<Box<dyn ::arrow2::array::Array>>
+    where
+        Self: 'a,
+    {
+        Self::try_to_arrow_opt(data.into_iter().map(Some))
+    }
+
+    /// Given an iterator of options of owned or reference values to the current
+    /// [`Loggable`], serializes them into an Arrow array.
+    /// The Arrow array's datatype will match [`Loggable::arrow_field`].
+    ///
+    /// Panics on failure.
+    /// This will _never_ fail for Rerun's built-in [`Loggable`].
+    ///
+    /// For the fallible version, see [`Loggable::try_to_arrow_opt`].
+    #[inline]
+    fn to_arrow_opt<'a>(
+        data: impl IntoIterator<Item = Option<impl Into<::std::borrow::Cow<'a, Self>>>>,
+    ) -> Box<dyn ::arrow2::array::Array>
+    where
+        Self: 'a,
+    {
+        Self::try_to_arrow_opt(data).detailed_unwrap()
+    }
+
+    // --- Optional deserialization methods ---
+
+    /// Given an Arrow array, deserializes it into a collection of [`Loggable`]s.
+    ///
+    /// Panics if the data schema doesn't match, or if optional entries were missing at runtime.
+    /// For the non-fallible version, see [`Loggable::try_from_arrow`].
+    #[inline]
+    fn from_arrow(data: &dyn ::arrow2::array::Array) -> Vec<Self> {
```

Comment:
Again, I find this a worrying footgun. The user can always call `.unwrap()` themselves if they feel adventurous